### DISCUSSION TITLE
How to use v3 generated client as a HTTP client library?

### DISCUSSION THREAD

**Discussion by touhonoob** [goadesign/goa#2994] [Q&A] [answered]
It seems unclear to me how to use the generated client code as a HTTP client library to goa service.
I figure there's a `NewClient` factory for creating clients generated for each goa service. Problem is there isn't any documentation/example on how to implement its `doer`, `enc`, `dec` params.
```
func NewClient(
	scheme string,
	host string,
	doer goahttp.Doer,
	enc func(*http.Request) goahttp.Encoder,
	dec func(*http.Response) goahttp.Decoder,
	restoreBody bool,
) *Client
```

**Comment by raphael** [comment, marked as answer]
The [clue](https://github.com/goadesign/clue) project includes a fully featured example comprised of three microservices: the `front` service which receives requests from the public internet via HTTP and the `forecaster` and `locator` services that serve gRPC requests coming from the `front` service (see [diagram](https://github.com/goadesign/clue/blob/main/example/weather/diagram/Weather%20System%20Services.svg)).

The `front` service implements [two client packages](https://github.com/goadesign/clue/tree/main/example/weather/services/front/clients), one for each service it makes requests to.  For example the `forecaster` client package contains a [Client struct](https://github.com/goadesign/clue/blob/main/example/weather/services/front/clients/forecaster/client.go#L56-L59) which exposes a [GetForecast](https://github.com/goadesign/clue/blob/main/example/weather/services/front/clients/forecaster/client.go#L70) method which makes a request to the downstream `forecaster` service.

**Reply by touhonoob** [reply]
thanks!
